invenio_records_resources/services/records/service.py

```diff
@@ -579,3 +581,74 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def create_or_update_many(self, identity, data, uow=None):
+        """Create or update a list of records.
+
+        This method takes a list of record data and creates or updates the corresponding records.
+
+        Args:
+            identity (object): The user identity performing the operation.
+            data (list): A list of tuples containing the record ID and record data.
+            uow (UnitOfWork, optional): The unit of work to register the record operations. Defaults to None.
+
+        Returns:
+            list: A list of tuples containing the operation type ('create' or 'update'), the processed record or the record dict, and any schema errors encountered.
+        """
+        records_processed = []
+        for record_id, record_dict in data:
```

Comment:
Hmm to use them we would need to change the logic and signature to be able to avoid saving the record in the DB when we have validation errors (and also pass the raise_errors=False), at the moment if there are errors it would either raise and stop or continue and save the record with the errors anyway. In the vocabs we only want to save the record if it succeed otherwise skip it and simply log it. 

If you consider that in the `RecordService` class the `create_or_update_many` should behave using the base `create` and `update` methods maybe we can simply override this method in the vocabs service. Let me know what you think we should do in this case.

Reply:
how is it done for the bulk members operation? if I remember correctly we were also gathering the errors from bulk operation in some data structure (not saying this might be needed here!)
is the requirement to abort and not save any records on the list if one fails or we should save only the correct one and omit the failed ones?

Reply:
The reason is because the default Service methods for `create`/`update` do direct indexing, while we want to send the records for bulk indexing. I think we should add a comment explaining this here, just so that it's a bit more explicit for anyone reading this part.